openpgp/packet/private_key.go

```diff
@@ -482,19 +531,61 @@ func (pk *PrivateKey) encrypt(passphrase []byte, blockCipher CipherFunction, s2k
 	return err
 }
 
-// encryptWithConfig encrypts an unencrypted private key using a passphrase and with the config.
-func (pk *PrivateKey) encryptWithConfig(passphrase []byte, c *Config) error {
-	return pk.encrypt(passphrase, c.Cipher(), c.S2K())
+// encryptWithConfig encrypts an unencrypted private key using the passphrase and the config.
+func (pk *PrivateKey) encryptWithConfig(passphrase []byte, config *Config) error {
+	params, err := s2k.Generate(config.Random(), config.S2K())
+	if err != nil {
+		return err
+	}
+	// Derive an encryption key with the configured s2k function.
+	key := make([]byte, config.Cipher().KeySize())
+	s2k, err := params.Function()
+	if err != nil {
+		return err
+	}
+	s2k(key, passphrase)
+	// Encrypt the private key with the derived encryption key.
+	return pk.encrypt(key, params, config.Cipher())
+}
+
+// EncryptPrivateKeys encrypts all unencrypted keys with the given config and passphrase.
+// Only derives one key from the passphrase, which is then used to encrypt each key.
+func EncryptPrivateKeys(keys []*PrivateKey, passphrase []byte, config *Config) error {
+	params, err := s2k.Generate(config.Random(), config.S2K())
+	if err != nil {
+		return err
+	}
+	// Derive an encryption key with the configured s2k function.
+	encryptionKey := make([]byte, config.Cipher().KeySize())
+	s2k, err := params.Function()
+	if err != nil {
+		return err
+	}
+	s2k(encryptionKey, passphrase)
+	for _, key := range keys {
+		if key != nil && !key.Dummy() && !key.Encrypted {
+			err = key.encrypt(encryptionKey, params, config.Cipher())
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
 }
 
 // Encrypt encrypts an unencrypted private key using a passphrase.
 func (pk *PrivateKey) Encrypt(passphrase []byte) error {
+	//Default config of private key encryption
 	s2kConfig := &s2k.Config{
 		S2KMode:  s2k.IteratedSaltedS2K,
 		S2KCount: 65536,
 		Hash:     crypto.SHA256,
+	} 
+	globalConfig := &Config{
+		S2KConfig: s2kConfig,
+		DefaultCipher: CipherAES256,
 	}
```

Comment:
OK, one more, these can be merged:
```suggestion
	// Default config of private key encryption
	config := &Config{
		S2KConfig: &s2k.Config{
			S2KMode:  s2k.IteratedSaltedS2K,
			S2KCount: 65536,
			Hash:     crypto.SHA256,
		} ,
		DefaultCipher: CipherAES256,
	}
```